File: Raspberry Pi/SmartMeter/EnergyData.py
from EnergyTariff import EnergyTariff

class EnergyData:
	ConsumedRate1 = -1
	ConsumedRate2 = -1
	ReturnedRate1 = -1
	ReturnedRate2 = -1
	Tariff = EnergyTariff.Unknown
	Consumed = -1
	Returned = -1
	Gas = -1

	def IsValid(self):
		return self.ConsumedRate1 >= 0 \
		and self.ConsumedRate2 >= 0 \
		and self.ReturnedRate1 >= 0 \
		and self.ReturnedRate2 >= 0 \
		and self.Tariff != EnergyTariff.Unknown \
		and self.Consumed >= 0 \
		and self.Returned >= 0 \
		and self.Gas >= 0

File: Web/Energy.DynamoDb/Models/EnergyData.cs
using Amazon.DynamoDBv2.DataModel;
using Energy.Core.Enumerations;
using Energy.Core.Interfaces;
using Energy.Repository.DynamoDb.Converters;
using System;

namespace Energy.Repository.DynamoDb.Models
{
    [DynamoDBTable("energy")]
    public class EnergyData : IEnergyData
    {
        [DynamoDBHashKey("deviceid")]
        public Guid DeviceId { get; set; }
        [DynamoDBRangeKey("timestamp", typeof(DateTimeConverter))]
        public DateTime Timestamp { get; set; }
        [DynamoDBProperty]
        public double ConsumedRate1 { get; set; }
        [DynamoDBProperty]
        public double ConsumedRate2 { get; set; }
        [DynamoDBProperty]
        public double ReturnedRate1 { get; set; }
        [DynamoDBProperty]
        public double ReturnedRate2 { get; set; }
        [DynamoDBProperty]
        public Tariff Tariff { get; set; }
        [DynamoDBProperty]
        public double Consumed { get; set; }
        [DynamoDBProperty]
        public double Returned { get; set; }
        [DynamoDBProperty]
        public double Gas { get; set; }
    }
}